lib/segment/tests/integration/payload_index_test.rs

```diff
@@ -795,3 +779,50 @@ fn test_update_payload_index_type() {
     assert_eq!(field_index[0].count_indexed_points(), point_num);
     assert_eq!(field_index[1].count_indexed_points(), point_num);
 }
+
+#[test]
+fn test_any_matcher_cardinality_estimation() {
+    let dir1 = Builder::new().prefix("segment1_dir").tempdir().unwrap();
+    let dir2 = Builder::new().prefix("segment2_dir").tempdir().unwrap();
+
+    let (struct_segment, _) = build_test_segments(dir1.path(), dir2.path());
+
+    let any_match = FieldCondition::new_match(
+        STR_KEY,
+        Match::new_any(AnyVariants::Keywords(vec![
+            "value1".to_string(),
+            "value2".to_string(),
+        ])),
+    );
+
+    let filter = Filter::new_must(Condition::Field(any_match.clone()));
+
+    let estimation = struct_segment
+        .payload_index
+        .borrow()
+        .estimate_cardinality(&filter);
+
+    let primary_clause = estimation.primary_clauses.first().unwrap();
+    match primary_clause {
+        PrimaryCondition::Condition(field_condition) => {
+            assert_eq!(field_condition, &any_match);
+        }
+        o => panic!("unexpected primary clause: {:?}", o),
+    }
+
+    let payload_index = struct_segment.payload_index.borrow();
+    let filter_context = payload_index.filter_context(&filter);
+    let exact = struct_segment
+        .id_tracker
+        .borrow()
+        .iter_ids()
+        .filter(|x| filter_context.check(*x))
+        .collect_vec()
+        .len();
+
+    eprintln!("exact = {exact:#?}");
+    eprintln!("estimation = {estimation:#?}");
+
+    assert!(exact <= estimation.max);
+    assert!(exact >= estimation.min);
```

Comment:
We cannot make this range smaller, can we? Since we implemented a proper cardinality estimation for `any` now.

Reply:
To validate the exact number, I would have to make sure my query is actually matching data in the test.
The data is generated randomly for the payloads, that's why we use those high level invariants in the tests.